src/mb_enhanced_cover_art_uploads/index.tsx

```diff
@@ -179,14 +180,14 @@ class ImageImporter {
         const originalFilename = url.pathname.split('/').at(-1) ?? 'image';
 
         // Prevent re-adding one we've already done
-        if (this.#doneImages.has(url)) {
+        if (this.#doneImages.has(url.href)) {
             this.#banner.set(`${originalFilename} has already been added`);
             this.#clearInput(url.href);
             return;
         }
-        this.#doneImages.add(url);
+        this.#doneImages.add(url.href);
 
-        let result;
+        let result: FetchResult;
```

Comment:
Do you happen to know of a setting that forces me to write types for such declarations? I get an eerily uncomfortable feeling seeing bare `let`s where the type of the variable isn't either explicitly specified or fairly easy to see from the initialiser.

Reply:
No idea, `"noImplicitAny": true` did not seem to have any effect on these, `tsc` did not even complain about them after relaunching the IDE...

Reply:
To be honest, I also have very little experince with TypeScript, but after having read through the possible `tsconfig.json` options, I also checked which options ESLint has to offer (including those of `@typescript-eslint`) and found the rule `'init-declarations': ['warn', 'always']` at least.
If we would enable this, it would at least trigger a warning for each declaration without initialisation, which makes it straight-forward to find the candidates for variables without inferred types or type annotations. Then we would have to mark them as exceptions:
```ts
// eslint-disable-next-line init-declarations
let result: FetchResult;
```
What do you think? Is it worth the ugly ESLint comments?

Thank you also very much for the invitation to become a collaborator and for the trust that I will not mess this repo up (I already added labels to this PR to test one of my new priviledges and failed at the first attempt due to the fat finger syndrome 😂).
If you want, I can push these changes (I see you are enjoying your experiments with the CI setup in a separate branch). I also have already prepared a commit that enables ESLint also for .jsx and .tsx files (after I wondered why my changes had no effect on the outcome of the linter).

Reply:
I always like a good hack like that, but I despise the ESLint ignore comment format (it's way too long, `flake8`'s `# noqa` ftw), so let's not 🙂

> I also have already prepared a commit that enables ESLint also for .jsx and .tsx files (after I wondered why my changes had no effect on the outcome of the linter).

Yes please. That explains why it wasn't complaining about [this unused import](https://github.com/ROpdebee/mb-userscripts/commit/32c5f0a0b44056f9b0cd8b46cc718bcb60aad7a0#diff-625513237b12a2d47cfd00590a1a8a2cefcb9cdbf2080df59c3e6ef614ab6a29L3) previously. I thought it had something to do with the `lib/*` ignore pattern (which is meant to ignore `./lib/`, not `./src/lib`), but you found the real cause

> Thank you also very much for the invitation to become a collaborator and for the trust that I will not mess this repo up

You're of course very welcome to contribute or help maintain the repo, but please don't feel pressured to do so. I added you as a collaborator so that if anything urgent needs to be addressed and I'm not around to do it for whatever reason, that at least one other person has write access, and you're the obvious choice. Don't interpret this as a nudge of "hey fix these bugs and implement these features" 😛 In regards to messing up the repo, I think we should still stick to PRs, and I'll be switching over to using feature branches and PRs myself too. I'll probably prohibit direct pushes to `main` as well to further enforce that. It's a better practice IMO, and I intend to eventually switch over to a Continuous Delivery system where a release is made on each PR merge, 1 PR = 1 feature/bug fix. It's pretty much what I'm mostly doing already anyway, but manually, so automating that seems like the right choice.

On that topic
> I see you are enjoying your experiments with the CI setup in a separate branch

Working on CI is so much fun, said no one ever. But it's going pretty well so far. The goal is to have a system that builds and commits to the `dist/` directory on each version increment in a `meta.ts` automatically, so that's a first step into the CD direction. Eventually it should increment versions itself though, but I'm not doing that until there are reliable tests

Reply:
Ok, so I will push the minor commit https://github.com/ROpdebee/mb-userscripts/commit/ade55719e0322fbd57638eb456b2f4484fa80a41 directly to `main` and we drop https://github.com/ROpdebee/mb-userscripts/commit/b96db12c8f22fe2b738a671e1a5b9767361fd2c8 , alright?